x-pack/plugins/files/common/api_routes.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PLUGIN_ID } from './constants';
+
+const API_BASE_PATH = `/api/${PLUGIN_ID}`;
+
+const FILES_API_BASE_PATH = `${API_BASE_PATH}/files`;
+
+export const FILE_KIND_API_ROUTES = {
+  getCreateFileRoute: (fileKind: string) => `${FILES_API_BASE_PATH}/${fileKind}`,
+  getUploadRoute: (fileKind: string) => `${FILES_API_BASE_PATH}/${fileKind}/{fileId}/blob`,
+  getDownloadRoute: (fileKind: string) =>
+    `${FILES_API_BASE_PATH}/${fileKind}/{fileId}/blob/{fileName?}`,
+  getUpdateRoute: (fileKind: string) => `${FILES_API_BASE_PATH}/${fileKind}/{fileId}`,
+  getDeleteRoute: (fileKind: string) => `${FILES_API_BASE_PATH}/${fileKind}/{fileId}`,
+  getListRoute: (fileKind: string) => `${FILES_API_BASE_PATH}/${fileKind}/list`,
+  getByIdRoute: (fileKind: string) => `${FILES_API_BASE_PATH}/${fileKind}/{fileId}`,
+};
```

Comment:
What do you think about consuming *file kind* form the path the same way as other arguments, by parsing the path? E.g.:

```suggestion
export const FILE_KIND_API_ROUTES = {
  getCreateFileRoute: `${FILES_API_BASE_PATH}/{fileKind}`,
  getUploadRoute: `${FILES_API_BASE_PATH}/{fileKind}/{fileId}/blob`,
  getDownloadRoute: `${FILES_API_BASE_PATH}/{fileKind}/{fileId}/blob/{fileName?}`,
  getUpdateRoute: `${FILES_API_BASE_PATH}/{fileKind}/{fileId}`,
  getDeleteRoute: `${FILES_API_BASE_PATH}/{fileKind}/{fileId}`,
  getListRoute: `${FILES_API_BASE_PATH}/{fileKind}/list`,
  getByIdRoute: `${FILES_API_BASE_PATH}/{fileKind}/{fileId}`,
};
```

The benefits I see:

- We register each path only once in the router.
- The code is simpler to understand, then the code with Proxy interception. We use a common pattern to extract path steps.
  - Now we can also log a custom error, if a request against unknown or illegal `fileKind` is executed.
- nit: `Proxy` objects are known to be slow.